### ISSUE TITLE
[JENKINS-70603] Infinite linking in pie charts

### ISSUE THREAD

**Issue by jenkins-infra-bot** [jenkinsci/warnings-ng-plugin#3107]
<p>Clicking on a pie chart, for example maven warnings' severities distribution and reference comparison creates infinite links stacking each one at the end, like</p>

<p><a href="https://ci.jenkins.io/job/Plugins/job/junit-plugin/job/master/lastSuccessfulBuild/maven-warnings/Normal/Normal/Normal/Normal/Normal/Normal/Outstanding/Outstanding/Outstanding/Outstanding/Outstanding/Outstanding/Normal/" class="external-link" target="_blank" rel="nofollow noopener">https://ci.jenkins.io/job/Plugins/job/junit-plugin/job/master/lastSuccessfulBuild/maven-warnings/Normal/Normal/Normal/Normal/Normal/Normal/Outstanding/Outstanding/Outstanding/Outstanding/Outstanding/Outstanding/Normal/</a></p>

<p> </p>

<p>I attached a video how to replicate the issue:</p>

<p><span class="nobr"><a href="https://issues.jenkins.io/secure/attachment/59966/59966_Kapture+2023-02-11+at+16.56.09.mp4" title="Kapture 2023-02-11 at 16.56.09.mp4 attached to <a class="jira-plain-text-key" href="https://issue-redirect.jenkins.io/issue/70603">JENKINS-70603</a>">Kapture 2023-02-11 at 16.56.09.mp4<sup><img class="rendericon" src="https://issues.jenkins.io/images/icons/link_attachment_7.gif" height="7" width="7" align="absmiddle" alt="" border="0"/></sup></a></span></p>

---
<details><summary><i>Originally reported by <img align="left" width="20" src="https://raw.githubusercontent.com/jenkinsci/artifacts-from-jira-issues/refs/heads/main/avatars/notmyfault.png" title="notmyfault's avatar" /> <a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=notmyfault">notmyfault</a>, imported from: <a class="original-jira-link" href="https://issues.jenkins.io/browse/JENKINS-70603" target="_blank">Infinite linking in pie charts</a></i></summary>
<i><ul>
<li><b>status</b>: Open
<li><b>priority</b>: Minor
<li><b>component(s)</b>: warnings-ng-plugin
<li><b>label(s)</b>: user-experience
<li><b>resolution</b>: Unresolved
<li><b>votes</b>: 0
<li><b>watchers</b>: 1
<li><b>imported</b>: 2025-12-09
</ul></i>
<details><summary>Raw content of original issue</summary>

<pre>
<p>Clicking on a pie chart, for example maven warnings' severities distribution and reference comparison creates infinite links stacking each one at the end, like</p>

<p><a href="https://ci.jenkins.io/job/Plugins/job/junit-plugin/job/master/lastSuccessfulBuild/maven-warnings/Normal/Normal/Normal/Normal/Normal/Normal/Outstanding/Outstanding/Outstanding/Outstanding/Outstanding/Outstanding/Normal/" class="external-link" target="_blank" rel="nofollow noopener">https://ci.jenkins.io/job/Plugins/job/junit-plugin/job/master/lastSuccessfulBuild/maven-warnings/Normal/Normal/Normal/Normal/Normal/Normal/Outstanding/Outstanding/Outstanding/Outstanding/Outstanding/Outstanding/Normal/</a></p>

<p> </p>

<p>I attached a video how to replicate the issue:</p>

<p><span class="nobr"><a href="https://issues.jenkins.io/secure/attachment/59966/59966_Kapture+2023-02-11+at+16.56.09.mp4" title="Kapture 2023-02-11 at 16.56.09.mp4 attached to JENKINS-70603">Kapture 2023-02-11 at 16.56.09.mp4<sup><img class="rendericon" src="https://issues.jenkins.io/images/icons/link_attachment_7.gif" height="7" width="7" align="absmiddle" alt="" border="0"/></sup></a></span></p></pre>
</details>
</details>
<ul><li><i>environment</i>: <code><a href="https://ci.jenkins.io/job/Plugins/job/junit-plugin/job/master/lastSuccessfulBuild/maven-warnings/">https://ci.jenkins.io/job/Plugins/job/junit-plugin/job/master/lastSuccessfulBuild/maven-warnings/</a></code></li></ul>
<details><summary><i>1 attachment</i></summary>

- [Kapture 2023-02-11 at 16.56.09.mp4](https://issues.jenkins.io/secure/attachment/59966/Kapture%202023-02-11%20at%2016.56.09.mp4)
</details>

**Comment by ayush18pop** [comment]
I traced down the issue and most likely the problem is in the echarts api plugin, In the webapp where renderPieCharts appends the params.name when clicking on chart 

the problem is in these lines of code (703-708) in (https://github.com/jenkinsci/echarts-api-plugin/blob/b318917fe90d95ea7b5642cb89b61d149eccbe99/src/main/webapp/js/echarts-api.js#L615)

 `const useLinks = chartPlaceHolder.attr('data-links');`
            `if (useLinks && useLinks !== "false") {`
                `chart.on('click', function (params) {`
                    `window.location.assign(params.name);`
               `});`
            `}`

**Comment by uhafner** [comment]
Yes, the property `useLinks` needs to be changed on sub pages. I never considered this problem as relevant so I did not change the implementation. I think with the new views in #2150 the whole concept needs to be restructured anyway so it makes not much sense to spend any effort here right now.
